src/index/inverted_index_reader.rs

```diff
@@ -351,11 +351,16 @@ impl InvertedIndexReader {
         let mut _term_info = self
             .get_term_range_async(.., automaton.clone(), None, merge_holes_under)
             .await?;
+
         // we build a 2nd iterator, this one with no holes, so we don't go through blocks we can't
         // match, and just download them to reduce our query count. This makes the assumption
         // there is a caching layer below, which might not always be true, but is in Quickwit.
         let term_info = self.get_term_range_async(.., automaton, None, 0).await?;
 
+        // TODO this operation is often cheap for "friendly" automatons, but can be very costly for
+        // "unfriendly" ones such as ".*a{50}" (very few terms if any  match this pattern, but we
+        // can't know early). In this case, we decompress and iterate over the entire sstable, while
+        // still being in async context. Ideally we should spawn this on a threadpool.
```

Comment:
I see. So if I understand correctly, the trouble is that the computation is done within an iterator. 

We eventually yield, but that's only when we effectively get an element.

Is decompressing a single block on the runtime is fast enough, we could probably write ugly code and have our iterator return `Option<Range<usize>>` to offer the runtime to yield more often.

If we do it this way, we might also want to actively yield.
I believe tokio only force yield on await, after await has been reach N times. (to be checked). That N might be way too high for us.

The second is probably not of a problem